File: Dreams/DreamServ/ccbAction/Models/Player.js
//Player Class - Definition of Properties for Player in Db
function Player(id, experience, level, health, mana, ownedByUserId, worldId, wPositionX, wPositionY, wPositionZ, inventoryId) {
    this.id = id;
    this.experience = experience;
    this.level = level;
    this.health = health;
    this.mana = mana;
    this.ownedByUserId = ownedByUserId;
    this.worldId = worldId;
    this.wPositionX = wPositionX;
    this.wPositionY = wPositionY;
    this.wPositionZ = wPositionZ;
    this.inventoryId = inventoryId;
}

//Getter-Functions for getting PlayerData

Player.prototype.getId = function() {
    this.id = ccbGetCopperCubeVariable("player_id");
}

Player.prototype.getHealth = function() {
    this.helath = ccbGetCopperCubeVariable("player_health"); 
}

Player.prototype.getExperiece = function () {
    this.experience = ccbGetCopperCubeVariable("player_xp");
}

Player.prototype.getLevel = function () {
    this.level = ccbGetCopperCubeVariable("player_level");
}

Player.prototype.getMana = function () {
    this.mana = ccbGetCopperCubeVariable("player_mana");
}

Player.prototype.getOwnerId = function () {
    this.ownedByUserId = ccbGetCopperCubeVariable("player_ownedById");
}

Player.prototype.getWorldId = function () {
    this.worldId = ccbGetCopperCubeVariable("player_worldId");
}

Player.prototype.getWpositionX = function () {
    this.getWpositionX = ccbGetCopperCubeVariable("player_posX");
}

Player.prototype.getWpositioY = function () {
    this.getWpositionY = ccbGetCopperCubeVariable("player_posY");
}

Player.prototype.getWpositionZ = function () {
    this.getWpositionZ = ccbGetCopperCubeVariable("player_posZ");
}


//Setter-Function for setting PlayerData

Player.prototype.setId = function () {
    ccbSetCopperCubeVariable("player_id", this.id);
}

Player.prototype.setHealth = function () {
    ccbSetCopperCubeVariable("player_health", this.health);
}

Player.prototype.setExperiece = function () {
    ccbSetCopperCubeVariable("player_Experience", this.experience);
}

Player.prototype.setLevel = function () {
    ccbSetCopperCubeVariable("player_level", this.level);
}

Player.prototype.setMana = function () {
    ccbSetCopperCubeVariable("player_mana", this.mana);
}

Player.prototype.setOwnerId = function () {
    ccbSetCopperCubeVariable("player_ownedById", this.ownedByUserId);
}

Player.prototype.setWorldId = function () {
    ccbSetCopperCubeVariable("player_worldId", this.worldId);
}

Player.prototype.setWpositionX = function () {
    ccbSetCopperCubeVariable("player_wPositionX", this.wPositionX);
}

Player.prototype.setWpositioY = function () {
    ccbSetCopperCubeVariable("player_wPositionX", this.wPositionX);
}

Player.prototype.setWpositionZ = function () {

}

File: Dreams/DreamServ/winDreamServ/_winDreamServ/Player.cs
namespace _winDreamServ
{
    internal class Player
    {

        public Player(int id, int experience, int mana, int armor, int health, int ownedByUserId, int worldId, int inventoryId, float wPositionX, float wPositionY, float wPositionZ)
        {
            this.Id = id;
            this.Experience = experience;
            this.Mana = mana;
            this.Armor = armor;
            this.Health = health;
            this.OwnedByUserId = ownedByUserId;
            this.WorldId = worldId;
            this.inventoryId = inventoryId;
            this.WPositionX = wPositionX;
            this.WPositionY = wPositionY;
            this.WPositionZ = wPositionZ;
        }
        private int Id { get => Id; set => Id = value; }
        private int Experience { get => Experience; set => Experience = value; }
        private int Mana { get => Mana; set => Mana = value; }
        private int Armor { get => Armor; set => Armor = value; }
        private int Health { get => Health; set => Health = value; }
        private int OwnedByUserId { get => OwnedByUserId; set => OwnedByUserId = value; }
        private int WorldId { get => WorldId; set => WorldId = value; }
        private int inventoryId { get => inventoryId; set => inventoryId = value; }
        private float WPositionX { get => WPositionX; set => WPositionX = value; }
        private float WPositionY { get => WPositionY; set => WPositionY = value; }
        private float WPositionZ { get => WPositionZ; set => WPositionZ = value; }
    }

}
